src/widgetastic/widget.py

```diff
@@ -1662,8 +1670,8 @@ def rows(self, *extra_filters, **filters):
             return self._filtered_rows(*extra_filters, **filters)
 
     def _all_rows(self):
-        for row_pos in range(len(self.browser.elements(self.ROWS, parent=self))):
-            row_pos = row_pos if not self._is_header_in_body else row_pos + 1
+        # passing index to TableRow, should not be <1 but want to adjust for headers within TableRow
+        for row_pos in range(1, len(self.browser.elements(self.ROWS, parent=self)) + 1):
```

Comment:
*REQUIRED* please re-write to:
```python
for row_pos in range(1, self.row_count + 1):
```

Same thing, easier to read :)

Reply:
_duh_ yeah good call.